tsconfig.eslint.json

```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "allowJs": true,
```

Comment:
Is the `allowJs` option necessary? I don't see more linting errors when I disable this option. Plus Babel has migrated to TS only so it is rare that we will ever import a .js from our source file.

I would like to minimize this config and hopefully in the future we can merge `tsconfig.eslint.json` back to `tsconfig.json` so we have consistent tsconfig between type checking and linting.